drakrun/drakrun/lib/postprocessing/capa_plugin/capa_processor.py

```diff
@@ -0,0 +1,452 @@
+    with metadata_path.open("r") as f:
+        # we use the metadata file to get the analysis' start command
+        metadata = msgspec.json.decode(f.read())
+
+    with inject_path.open("r") as f:
+        # we use the injected processes log to get the malware process' parent pid
+        injected_processes = [msgspec.json.decode(line) for line in f]
+
+    with pstree_path.open("r") as f:
+        # we use the process tree to get all of the malware process' child processes
+        pstree = msgspec.json.decode(f.read())
+
+    # make sure we have the right inject.log entry (using the metadata.json file)
+    malware_injection_log: str = next(
+        filter(
+            lambda line: line["ProcessName"] == metadata["start_command"],
+            injected_processes,
+        )
+    )
+
+    # get the parent malware process' pid and process name
+    malware_pid: int = malware_injection_log["InjectedPid"]
+    malware_procname: str = malware_injection_log["ProcessName"]
+
+    # find the malware process' sub-tree in the overall process tree
+    try:
+        malware_process: Dict = find_process_in_pstree(
+            pstree, malware_pid, malware_procname
+        )
+    except ValueError as e:
+        # malware pid not found, analyze entire log instead
+        return None
+
+    # get all child processes of the malware process
+    malware_pids = get_all_child_processes(malware_process)
+
+    # return a list of all spawned malware processes (does not include process injection)
+    return malware_pids
+
+
+def get_drakvuf_feature_extractor(
+    calls: Iterator[Dict],
+) -> capa.features.extractors.drakvuf.extractor.DrakvufExtractor:
+
+    # wrapper routing for initializing the Drakvuf feature extractor
+    return capa.features.extractors.drakvuf.extractor.DrakvufExtractor.from_report(
+        calls
+    )
+
+
+def decode_json_lines(fd: BinaryIO) -> Iterator[Dict]:
+    # read and decode json lines into a list of dictionaries
+    for line in fd:
+        try:
+            # we use msgspec for a small performance improvement
+            line_s = line.strip().decode()
+            obj = msgspec.json.decode(line_s)
+            yield obj
+        except (msgspec.DecodeError, UnicodeDecodeError):
+            # sometimes Drakvuf reports bad method names and/or malformed JSON
+            logger.debug("bad drakvuf log line: %s", line)
+
+
+def dynamic_capa_analysis(
+    analysis_dir: pathlib.Path,
+    rules: capa.rules.RuleSet,
+    malware_pids: Optional[List[int]] = None,
+) -> Tuple[pathlib.Path, capa.render.result_document.MatchResults]:
+
+    # save all api calls and native calls into one list which gets sorted by capa later on.
+    calls = []
+
+    # read api calls
+    try:
+        with (analysis_dir / "apimon.log").open("rb") as apimon_fd:
+            calls += list(decode_json_lines(apimon_fd))
+    except:
+        logger.debug("empty apimon.log file")
```

Comment:
Should I get rid of all uses of `try: ... except: ...` in the code? cause in some instances it can be a bit tricky (the capa part for example, since it can throw some exceptions).